### PR TITLE
Sortable HeadCell

### PR DESCRIPTION
**Pull request by Joozty**

### CODE REVIEW

**Comment by viktor-yakubiv** [comment]
I prefer having explicit button for it. We need to use aria attributes to handle clicks, focus and so on. It's not very convenient.

Also, once you push this API it will be really hard to get rid out of.

**Comment by Joozty** [comment]
Not sure if I understand. Which one?

1. A button will have `::after` element with arrow up/down
```
<HeadCell>
	<Button>{children}</Button>
</HeadeCell>
```

2. A button will contain a clickable icon
```
<HeadCell>
	{children}
	<Button>
		<Icon />
	</Button>
</HeadeCell>
```

3. A button will contain children with arrow icon.
```
<HeadCell>
	<Button>
		{children}
		<Icon />
	</Button>
</HeadeCell>
```

**Comment by viktor-yakubiv** [comment]
Having kind of following API

```jsx
<Table.HeadCell>
  Title
  <Table.SortButton onClick={handler}>Sort by title</Table.SortButton>
</Table.HeadCell>
```

**Comment by Joozty** [comment]
@viktor-yakubiv I was thinking about it and I decided to now remove `:after` element in favour of an icon. I wanted to have whole cell clickable (because of mobile devices) and I found it more complicated to implement. Also, this is a more native way I think. I made is as much accessible as possible. Please have a look. 🙂

File: src/elements/tables/head-cell.jsx [hunk 1]
```diff
@@ -2,33 +2,67 @@ import React from 'react'
...
+  ) => (
     <Tag
       className={classNames
-        .use({ sticky })
+        .use({
+          sticky,
+          [styles.headCell]: true,
+          [styles.sortable]: sortable,
+          [styles.asc]: order === 'ascending',
+          [styles.desc]: order === 'descending',
```
[the remark this answers is not in the document]
Comment (Joozty) [reply src/elements/tables/head-cell.jsx]: true thanks

**Comment by Joozty** [comment]
@viktor-yakubiv I was trying to discover how the other libraries do it. Everybody does it differently. For example, [material ui](https://material-ui.com/components/tables/) does it like this:

```
<th>
<span role="button"...>
	<span classname="sr-only">Sorted ascending</span>
	<svg focusable="false" aria-hidden="true" />
</span>
</th>
``` 

Anyway, I also tried screen reader and effect is the same for solution present in this pull request and for solution in material ui. 


>  Currently, when you focus on button, screen reader would tell you something like 'button, caption: Title'.

It says `Name of column`, `not sortable|sorted ascending|sorted descending`, button, column header

I think it's correct.

**Comment by viktor-yakubiv** [comment]
Thank you @Joozty, especially for this investigation of screen reader's behaviour 👍

**Pull request merged**